### ISSUE TITLE
 CPU usage increase in Sql Servers when migrating from v3 to latest

### ISSUE THREAD

**Issue by johnsimons** [OctopusDeploy/Issues#4681]
We have multiple reports of an increase in Sql Servers CPU usage after upgrading from v3 to latest.

The latest report seems to indicate the spikes occur during deployments.

**Symptoms:**
Large increase in CPU usage on SQL Server, often pegging the CPU at 100%.  This is especially prevalent during peak hours. Reports suggest this is happening most when deployments are in progress.

**Upgrade paths:**
3.17.14 -> 2018.6.5
3.17.4 -> 2018.3.4
3.17.14 -> 2018.4.10

**Comment by droyad** [comment]
Issues introduced sometime after `3.17.14`

**Comment by tothegills** [comment]
Sources:
https://secure.helpscout.net/conversation/605703012/28338/?folderId=557080
https://help.octopus.com/t/server-performance-slow-since-update-from-3-17-4-to-2018-3-4/19491
https://secure.helpscout.net/conversation/603112656?folderId=557082
https://secure.helpscout.net/conversation/595133164?folderId=557082

**Comment by tothegills** [comment]
Stats from one report:

SQL CPU usage before and after upgrade from 3.17.14 -> 2018.6.5:
![image](https://user-images.githubusercontent.com/171197/42002265-fa6f46c8-7aa9-11e8-9d10-f8334f9aed7f.png)

Long-running request count:
![image](https://user-images.githubusercontent.com/171197/42002344-576b9692-7aaa-11e8-88d0-27f5ae6d7e27.png)

Long-running dashboard requests:
![image](https://user-images.githubusercontent.com/171197/42002373-6eb9ab0e-7aaa-11e8-97ee-498491679215.png)

Long-running progression requests:
![image](https://user-images.githubusercontent.com/171197/42002388-804d0974-7aaa-11e8-9d92-ce887ebe286b.png)

**Comment by fenneh** [comment]
I'm not sure if this should be expanded a little but I have noticed a massive performance hit in Octopus and it's SQL Queries since upgrading from 3.17.10 to 2018.3.4 and all versions since. I know there's been improvements made in this area but personally I'm still seeing long page loads on dashboard/project pages. 

I've also wondered if it was SQL Server's cache being cold but that's not the case, have also migrated to SQL 2017 and to  a new box and still the sluggishness is there. The only thing I have left is to apply my own indexes and hope it doesn't cause an issue with upgrades (or pull and re-add them)

Am happy to send across all logs and information.

**Comment by fenneh** [comment]
We're running 2018.6.13; 

Here's some examples of the load times we're seeing 

```
	July 12th 2018, 10:04:14.941	Request took 9285ms: GET http://octopus/api/progression/Projects-762 4b26eae337d644e4b5931c154e6f8a09 Redacted
	July 12th 2018, 10:04:14.941	Request took 9471ms: GET http://octopus/api/progression/Projects-848 d99544ade250425c917cd27563bfed8e Redacted
	July 12th 2018, 10:04:14.941	Request took 10816ms: GET http://octopus/api/progression/Projects-100 a086acb1979441bca0edc60dda6bd48b Redacted
	July 12th 2018, 10:04:14.941	Request took 12487ms: GET http://octopus/api/progression/Projects-605 1dea545b91a742db8ea59f778eb99b39 Redacted
	July 12th 2018, 09:52:08.801	Request took 9499ms: GET http://octopus/api/progression/Projects-848 6868cda9283a44e7abf6ad668adbac6d Redacted
	July 12th 2018, 09:52:08.801	Request took 9639ms: GET http://octopus/api/progression/Projects-100 94cbb679319742cb91b253b58a10f107 Redacted
	July 12th 2018, 09:52:08.801	Request took 10183ms: GET http://octopus/api/progression/Projects-275 c82aa2fd58a34ec894816d7021703bf1 Redacted
	July 12th 2018, 09:50:21.778	Request took 15741ms: GET http://octopus/api/progression/Projects-275 c267cc055abd42b2a8c23380c42c7594 Redacted
	July 12th 2018, 09:50:21.778	Request took 14942ms: GET http://octopus/api/progression/Projects-848 f997f579aa3f467dbc5c34bab46cce5b Redacted
	July 12th 2018, 09:50:21.778	Request took 11135ms: GET http://octopus/api/progression/Projects-402 a278ef91445d481585f76c5b59fe4663 Redacted
	July 12th 2018, 09:50:21.778	Request took 15774ms: GET http://octopus/api/progression/Projects-100 55a298480dfe45028caf6b47b3f7a76d Redacted
	July 12th 2018, 09:50:21.778	Request took 12633ms: GET http://octopus/api/progression/Projects-605 2bb0f6aad4c448408e69a127f6262b12 Redacted
	July 12th 2018, 09:50:21.778	Request took 14749ms: GET http://octopus/api/progression/Projects-762 6ed4d490d951438db5387f6ee4c31eac Redacted
	July 12th 2018, 09:49:15.720	Request took 11226ms: GET http://octopus/api/progression/Projects-848 32519cc83a9943708f3511b9bc2bc0b3 Redacted
	July 12th 2018, 09:49:15.720	Request took 11489ms: GET http://octopus/api/progression/Projects-762 df92b63f51bd4afd83fe146cb4b878ed Redacted
	July 12th 2018, 09:49:15.720	Request took 5740ms: GET http://octopus/api/progression/Projects-689 d07d2a06c42c4bb7822ea0cedd4c745c Redacted
	July 12th 2018, 09:49:14.718	Request took 13681ms: GET http://octopus/api/progression/Projects-100 ae8fb560e0cf45a5b5ab4bd4751bcf7d Redacted
	July 12th 2018, 09:49:13.718	Request took 14314ms: GET http://octopus/api/progression/Projects-275 3ed1dcf1ccd7403fa8c5bfd53ac3d79f Redacted
	July 12th 2018, 09:46:23.689	Request took 11530ms: GET http://octopus/api/progression/Projects-605 b57104fe5bdf4a4a9da8a37b8738f4e1 Redacted
	July 12th 2018, 09:46:23.689	Request took 11646ms: GET http://octopus/api/progression/Projects-402 020483b20474460fb027e7acca03c801 Redacted
	July 12th 2018, 09:46:23.689	Request took 10984ms: GET http://octopus/api/progression/Projects-275 fc915f2c834545778aefa1b501df5df8 Redacted
	July 12th 2018, 09:46:23.689	Request took 11069ms: GET http://octopus/api/progression/Projects-762 d08cf5b7ef734bfdb6e6d33df0c27838 Redacted
	July 12th 2018, 09:46:23.689	Request took 11068ms: GET http://octopus/api/progression/Projects-848 9231159df22b47fcaef7a456c28a9ab4 Redacted
```

I've filtered out requests which took under 9 seconds just from a tiny slice of this morning.

I did originally bring this up in Slack and I wrongly came to the idea that it was our SQL cache not being populated.

**Comment by tothegills** [comment]
Logs from @fenneh: https://secure.helpscout.net/conversation/621434669/29400/?folderId=571723 (private link)

**Comment by droyad** [comment]
Customers are reporting are that this has been resolved in `2017.7.7`. The resolution was spread out across various issues (including #4718, #4748, #4761) and releases.
